src/IssueViz.UnitTests/Commands/IssueVisualizationToolWindowCommandTests.cs

```diff
@@ -0,0 +1,165 @@
+using SonarLint.VisualStudio.IssueVisualization.IssueVisualizationControl;
+using ThreadHelper = SonarLint.VisualStudio.Integration.UnitTests.ThreadHelper;
+
+namespace SonarLint.VisualStudio.IssueVisualization.UnitTests.Commands
+{
+    [TestClass]
+    public class IssueVisualizationToolWindowCommandTests
+    {
+        private Mock<AsyncPackage> package;
+        private Mock<IMenuCommandService> commandService;
+        private Mock<ILogger> logger;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            package = new Mock<AsyncPackage>();
+            commandService = new Mock<IMenuCommandService>();
+            logger = new Mock<ILogger>();
+
+            ThreadHelper.SetCurrentThreadAsUIThread();
+        }
+
+        [TestMethod]
+        public void Ctor_ArgsCheck()
+        {
+            Action act = () => new IssueVisualizationToolWindowCommand(null, commandService.Object, logger.Object);
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("package");
+
+            act = () => new IssueVisualizationToolWindowCommand(package.Object, null, logger.Object);
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("commandService");
+
+            act = () => new IssueVisualizationToolWindowCommand(package.Object, commandService.Object, null);
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("logger");
+        }
+
+        [TestMethod]
+        public void Ctor_CommandAddedToMenu()
+        {
+            new IssueVisualizationToolWindowCommand(package.Object, commandService.Object, logger.Object);
+
+            commandService.Verify(x =>
+                    x.AddCommand(It.Is((MenuCommand c) =>
+                        c.CommandID.Guid == IssueVisualizationToolWindowCommand.CommandSet &&
+                        c.CommandID.ID == IssueVisualizationToolWindowCommand.CommandId)),
+                Times.Once);
+        }
+
+        [TestMethod]
+        public void Execute_WindowNotCreated_NoException()
+        {
+            SetupWindowCreation(null);
+
+            VerifyExecutionDoesNotThrow();
+
+            VerifyMessageLogged("cannot create window frame");
+        }
+
+        [TestMethod]
+        public void Execute_WindowCreatedWithoutFrame_NoException()
+        {
+            SetupWindowCreation(new ToolWindowPane());
+
+            VerifyExecutionDoesNotThrow();
+
+            VerifyMessageLogged("cannot create window frame");
+        }
+
+        [TestMethod]
+        public void Execute_ExceptionInCreatingWindow_NoException()
+        {
+            SetupWindowCreation(new ToolWindowPane {Frame = Mock.Of<IVsWindowFrame>()}, new NotImplementedException("this is a test"));
+
+            VerifyExecutionDoesNotThrow();
+
+            VerifyMessageLogged("this is a test");
+        }
+
+        [TestMethod]
+        public void Execute_WindowCreatedWithIVsWindowFrame_WindowIsShown()
```

Comment:
It's actually not possible to create a window with a non-IVsWindowFrame, an exception is thrown by VS `new ToolWindowPane { Frame = new object() }` that it's not an IVsWindowFrame. So that's why there's no test for that.